psydac/fem/partitioning.py

```diff
@@ -54,10 +54,10 @@ def partition_coefficients(domain_decomposition, spaces, min_blocks=None):
         min_blocks = [None]*ndims
 
     for s,e,V,mb in zip(global_starts, global_ends, spaces, min_blocks):
-        if mb is None:
+        if V.periodic or mb is None:
             assert all(e-s+1>=V.degree)
         else:
-            assert all(e-s+1>=mb*(1-V.periodic)+V.degree*V.periodic)
+            assert all(e-s+1>=mb*(1-V.periodic))
```

Comment:
If you write it like this, you only require 1 coefficient per process in the periodic case. Is this really what you want?

Reply:
By the way, you cannot enter this branch in the periodic case! So you can just write what I suggested earlier...

Reply:
ah sorry, you are right , fixed in [3065ee1](https://github.com/pyccel/psydac/pull/240/commits/3065ee120d156251f3545f252de400eca0a20758)

Reply:
In the periodic case we were not gonna enter that part, so the code still runs, but it's not necessary to keep the condition as it was because we always have `(1-V.periodic) == 1`